Cython/Compiler/PyrexTypes.py

```diff
@@ -3844,10 +3851,14 @@ def create_to_py_utility_code(self, env):
                 'maybe_unordered': self.maybe_unordered(),
                 'type': self.cname,
             })
+            directives = dict(env.directives)
+            # if present, cpp_locals will invalidate some utility code
+            # (TODO filter directives more thoroughly to keep only those that are relevant)
+            directives["cpp_locals"] = False
```

Comment:
Since we will definitely have to touch this code again (possibly with a dict comprehension), let's keep it simple.
```suggestion
            # Override directives that should not be inherited from user code.
            # TODO: filter directives with an allow list to keep only those that are safe and relevant.
            directives = dict(env.directives, cpp_locals=False)
```